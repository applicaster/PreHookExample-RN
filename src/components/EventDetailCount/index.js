import React from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  Text,
  View,
 } from 'react-native';
import hexToRgb from 'hex-to-rgb';

const styles = StyleSheet.create({
  eventDetailCount: {
    flexDirection: 'row',
    marginRight: 5,
  },
  count: {
    fontSize: 12,
    marginRight: 3,
  },
  label: {
    fontSize: 12,
  },
});

const EventDetailCount = ({ label, count = 0 }, { textColor }) => {
  const rgb = hexToRgb(textColor || '#FFFFFF');
  const rgbaColor = `rgba(${rgb[0]},${rgb[1]},${rgb[2]}, 0.60)`;
  const textColorStyle = { color: rgbaColor };
  return (
    <View style={styles.eventDetailCount}>
      <Text key={'count'} style={[styles.count, textColorStyle]}>{count}</Text>
      <Text key={'label'} style={[styles.label, textColorStyle]}>{label}</Text>
    </View>
  );
};

EventDetailCount.propTypes = {
  label: PropTypes.string,
  count: PropTypes.number,
};

EventDetailCount.contextTypes = {
  textColor: PropTypes.string,
};

export default EventDetailCount;
