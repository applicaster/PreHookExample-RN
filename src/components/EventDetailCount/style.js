import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  eventDetailCount: {
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 5,
  },
  count: {
    fontSize: 12,
    marginRight: 3,
  },
  label: {
    fontSize: 12,
  },
});
