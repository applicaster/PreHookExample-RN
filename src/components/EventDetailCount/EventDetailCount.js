import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import hexToRgb from 'hex-to-rgb';
import { styles } from './style';

class EventDetailCount extends Component {
  constructor(props) {
    super(props);
    this.navigateToComments = this.navigateToComments.bind(this);
  }

  navigateToComments() {
    const { navigation } = this.context;
    const { eventId, setActiveEventId } = this.props;

    setActiveEventId(eventId);

    if (navigation.state.routeName !== 'Comments') {
      navigation.navigate('Comments');
    }
  }

  render() {
    const { label, count = 0 } = this.props;
    const { textColor } = this.context;
    const rgb = hexToRgb(textColor || '#FFFFFF');
    const rgbaColor = `rgba(${rgb[0]},${rgb[1]},${rgb[2]}, 0.60)`;
    const textColorStyle = { color: rgbaColor };
    return (
      <TouchableWithoutFeedback onPress={this.navigateToComments}>
        <View style={styles.eventDetailCount}>
          <Text key={'count'} style={[styles.count, textColorStyle]}>{count}</Text>
          <Text key={'label'} style={[styles.label, textColorStyle]}>{label}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

EventDetailCount.propTypes = {
  count: PropTypes.number,
  eventId: PropTypes.string,
  label: PropTypes.string,
  setActiveEventId: PropTypes.func,
};

EventDetailCount.contextTypes = {
  textColor: PropTypes.string,
  navigation: PropTypes.object,
};

export default EventDetailCount;
