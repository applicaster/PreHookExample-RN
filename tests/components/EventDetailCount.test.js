import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventDetailCount from '../../src/components/EventDetailCount/EventDetailCount';

describe('EventDetailCount', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <EventDetailCount count={1} label={'Likes'} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly with null data', () => {
    const html = renderer.create(
      <EventDetailCount count={null} label={null} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
