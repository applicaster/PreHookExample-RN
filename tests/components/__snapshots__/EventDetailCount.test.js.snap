// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailCount component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginRight": 5,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
          "marginRight": 3,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  >
    1
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  >
    Likes
  </Text>
</View>
`;

exports[`EventDetailCount component renders correctly with null data 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginRight": 5,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
          "marginRight": 3,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  />
</View>
`;
